onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -227,6 +227,12 @@ def run(self, output_names, input_feed, run_options=None):
                 return self._sess.run(output_names, input_feed, run_options)
             raise
 
+    def run_async(self, output_names, input_feed, callback, run_options=None):
```

Comment:
I would add a detailed docstring

Reply:
Would be great to have type annotations too in the function def, for clarity

Reply:
Added.